lib/Transforms/Scalar/DxilConditionalMem2Reg.cpp

```diff
@@ -356,6 +484,7 @@ class DxilConditionalMem2Reg : public FunctionPass {
 
     Changed |= RewriteOutputArgsDebugInfo(F);
     Changed |= dxilutil::DeleteDeadAllocas(F);
+    Changed |= SplitSimpleAllocas(F);
```

Comment:
I originally wondered if we were missing an opportunity by placing this after dead alloca deletion, but I think that no unused allocas will be created by the split, so none would be deleted.